check_certs.ps1

```diff
@@ -0,0 +1,74 @@
+##########################################################################################
+#
+# Name:        check_certs.ps1
+# Date:        2023-11-15
+# Author:      stefan.petronijevic@atomia.com
+# Version:     1.1.0
+# Parameters:  Without parameters, default values 15 and 30 are used for warning and
+#              critical, respectively. Custom values can be used.
+#              
+#              .\check_certs.ps1 10 20
+#                  10 - value for WARNING
+#                  20 - value for CRITICAL
+#
+# Returns:
+#              0 - OK
+#              1 - WARNING
+#              2 - CRITICAL
+#
+# Description: Nagios plugin that alerts whenever some certificate from store is due for
+#              renewal, and alerts based on the number of days until the expiration.
+#
+##########################################################################################
+
+param(
+    [int]$CriticalThreshold = 15, # Default value if not specified
+    [int]$WarningThreshold = 30   # Default value if not specified
+)
+
+# Initialize the exit code - 0 means no certificates are near expiration
+$exitCode = 0
+
+# Check if warning threshold is less than critical threshold and swap if necessary
+if ($WarningThreshold -lt $CriticalThreshold) {
+    $temp = $WarningThreshold
+    $WarningThreshold = $CriticalThreshold
+    $CriticalThreshold = $temp
+}
+
+# Specify the certificate store to check
+$storeLocation = "Cert:\LocalMachine\My"

```

Comment:
The certificate store location should also be a parameter since we could have this script check Root certificates which are in a different location or some other location for certificates.

You should still keep the current location `Cert:\LocalMachine\My` as a default on the argument that will be added.